src/matchers/toThrowWithMessage/index.js

```diff
@@ -0,0 +1,98 @@
+import { matcherHint, printExpected, printReceived } from 'jest-matcher-utils';
+
+import predicate from './predicate';
+
+const positiveHint = matcherHint('.toThrowWithMessage', 'function', 'type', { secondArgument: 'message' });
+const negativeHint = matcherHint('.not.toThrowWithMessage', 'function', 'type', { secondArgument: 'message' });
+
+const passMessage = (received, expected) => () => `
+${negativeHint}
+
+Expected not to throw: 
+  ${printExpected(expected)}
+Thrown:
+  ${printReceived(received)}
+`;
+
+const failMessage = (received, expected) => () => `
+${positiveHint}
+
+Expected to throw:
+${printExpected(expected)}
+Thrown:
+${printReceived(received)}
+`;
+
+export default {
+  toThrowWithMessage: (callback, type, message) => {
+    if (!callback || typeof callback !== 'function') {
+      return {
+        pass: false,
+        message: () => `
```

Comment:
The spacing of these validation errors is tabbed in too much for the printouts (see the snapshots)

Reply:
Alright, I've pushed in a change for this. Do take a look.